enhancements/installer/crc.md

```diff
@@ -0,0 +1,93 @@
+---
+title: code-ready-containers
+authors:
+reviewers:
+approvers:
+creation-date: 2020-03-02
+last-updated: 2020-03-02
+status: implemented
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+superseded-by:
+---
+
+# Code Ready Containers (CRC)
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+CRC functions by doing roughly....
+ 1.  Running a libvirt install, which is for ["development only"](https://github.com/openshift/installer/blob/ffc34e32fe4a71560f59312384daa87b401d6ec9/README.md) meaning OpenShift development only.
+ 2.  On a single node cluster, which is below the minimum of [three that are required](https://github.com/openshift/installer/blob/c904277e59dd947a8884265b2511034b05c38644/upi/openstack/inventory.yaml#L35) for a functional cluster.
+ 3.  Disable the CVO, which prevents all reconciliation of operators, top level status, and upgrades.
+ 4.  Deletes the monitoring stack, but not all monitor resources.
+     Removes the aggregated apiserver as well.
+ 5.  Scale insights to zero.
+ 6.  Force the router down to a single node.  Preventing smooth upgrades.
+ 7.  Twiddle the registry operator.
+     Changes for routes and PVs.
```

Comment:
What should be changed for routes?

Reply:
@dmage routes changes are https://github.com/code-ready/snc/blob/master/snc.sh#L265 , I think this is supported one.

Reply:
@dmage @praveenkumar enabling the default public route is the extra bit CRC does (a fully supported day 2 operation).

https://github.com/code-ready/snc/blob/master/snc.sh#L268

Reply:
Requesting 100Gi of storage feels like a tall order for a laptop [1]. @praveenkumar Does the CSI driver used by CRC support volume expansion[2]?

[1] https://github.com/code-ready/snc/blob/master/registry_pvc.yaml 
[2] https://kubernetes.io/docs/concepts/storage/persistent-volumes/#csi-volume-expansion

Reply:
Requesting and getting is a very different thing. At the moment the disks are limited in size to about 40G. Since we are using RHCOS the underlying filesystem is XFS, but there is no way to scale this out as a diskimage as the hypervisors might not support this anyway.